packages/argo-run/src/dev.ts

```diff
@@ -89,18 +91,115 @@ export async function dev(...args: string[]) {
+                  }
+
+                  body {
+                    min-height: 100vh;
+                    min-height: -webkit-fill-available;
+                    display: grid;
+                    margin: 0;
+                    align-content: center;
+                    justify-content: center;
+
+                    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
+                      'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
+                      sans-serif;
+                  }
+
+                  html {
+                    height: -webkit-fill-available;
+                    background: black;
+                    color: white;
+                  }
+
+                  .content {
+                    max-width: 40rem;
+                    padding: 2rem;
+                    font-size: 1.5em;
+                  }
+
+                  code {
+                    color: mediumseagreen;
+                    word-break: keep-all;
+                  }
+                </style>
+              </head>
+
+              <body>
+                <div class="content">
+                  <p>
+                    This page is served by your local Argo development server. Instead of
+                    visiting this page directly, you will need to connect your local development
+                    environment to a real checkout environment.
+                  </p>
+                  <p>${content}</p>
+                </div>
+              </body>
+            </html>
+          `);
+        }
+
+        if (protocol === 'https') {
+          renderIndexPage(
+            `Create a checkout and append <code>?dev=${origin}/query</code> to the URL to start developing your extension.`,
+          );
+        } else {
+          renderIndexPage(`Make sure you have an secure URL for your local development server by running <code>shopify tunnel start --port=${port}</code>,
+            create a checkout, and append <code>?dev=https://TUNNEL_URL/query</code> to the URL, where <code>TUNNEL_URL</code> is
+            replaced with your own ngrok URL.`);
+        }
+
         res.end();
       });
 
+      // This endpoint is used by C1 to request information about the local workspace,
+      // including what extensions are available locally, and how to connect to the
+      // build server websocket for those extensions.
+      app.get('/query', (req, res) => {
+        const protocol = req.headers['x-forwarded-proto'] ?? req.protocol;
+        const host = req.headers.host;
+        const origin = `${protocol}://${host}`;
+
+        res.set('Access-Control-Allow-Origin', '*');
+        res.set('Cache-Control', 'no-cache');
+
+        res.json({
+          queryUrl: `${origin}${req.path}`,
+          extensions: [
+            {
+              scriptUrl: `${origin}${PUBLIC_PATH}${filename}`,
+              socketUrl: `${origin.replace(/^http/, 'ws')}${WEBSOCKET_PATH}`,
```

Comment:
Should the manifest fields be named the same as Admin? https://github.com/Shopify/argo-admin/blob/1c977dfa7a261f083e01a56562a4dc012cc99544/packages/argo-admin-cli/src/server/index.ts#L83

Reply:
I don't think it matters, personally. The shape you have in the admin is clearly too limited for the next step (multiple extensions in one project, so I wanted to jump ahead a bit and get a more flexible manifest shape. The fields could be made the same, but considering the host code will be custom for each right now, I am not too worried about that.